mne/coreg.py

```diff
@@ -1245,3 +1248,558 @@ def _scale_xfm(subject_to, xfm_fname, mri_name, subject_from, scale,
+                len(self._nearest_transformed_high_res_mri_idx_hpi) > 0)
+
+    @property
+    def _has_eeg_data(self):
+        return (self._has_mri_data and
+                len(self._nearest_transformed_high_res_mri_idx_eeg) > 0)
+
+    @property
+    def _has_lpa_data(self):
+        return (np.any(self.lpa) and np.any(self._dig['lpa']))
+
+    @property
+    def _has_nasion_data(self):
+        return (np.any(self.nasion) and np.any(self._dig.nasion))
+
+    @property
+    def _has_rpa_data(self):
+        return (np.any(self.rpa) and np.any(self._dig['rpa']))
+
+    @property
+    def _processed_high_res_mri_points(self):
+        return self._get_processed_mri_points('high')
+
+    @property
+    def _processed_low_res_mri_points(self):
+        return self._get_processed_mri_points('low')
+
+    def _get_processed_mri_points(self, res):
+        bem = self.bem_low_res if res == 'low' else self.bem_high_res
+        if self._grow_hair:
+            if len(bem['nn']):
+                scaled_hair_dist = (1e-3 * self._grow_hair /
+                                    np.array(self._scale))
+                points = bem['rr'].copy()
+                hair = points[:, 2] > points[:, 1]
+                points[hair] += bem['nn'][hair] * scaled_hair_dist
+                return points
+            else:
+                raise ValueError("Norms missing from bem, can't grow hair")
+                self._grow_hair = 0
+        else:
+            return bem['rr']
+
+    @property
+    def _has_mri_data(self):
+        return len(self._transformed_high_res_mri_points) > 0
+
+    @property
+    def _has_dig_data(self):
+        return (self._has_mri_data and
+                len(self._nearest_transformed_high_res_mri_idx_hsp) > 0)
+
+    @property
+    def _orig_hsp_point_distance(self):
+        mri_points = self._transformed_high_res_mri_points[
+            self._nearest_transformed_high_res_mri_idx_orig_hsp]
+        hsp_points = self._transformed_orig_dig_extra
+        return np.linalg.norm(mri_points - hsp_points, axis=-1)
+
+    def _log_dig_mri_distance(self):
+        errs_nearest = dig_mri_distances(self._info, self.trans, self._subject,
+                                         self._subjects_dir)
+        logger.info('Median distance from digitized points to head surface '
+                    'using nearest neighbor is %.3f mm' % np.median(
+                        errs_nearest * 1000))
+
+    @verbose
+    def fit_fiducials(self, lpa_weight=1., nasion_weight=10., rpa_weight=1.,
+                      verbose=None):
+        """Find rotation and translation to fit all 3 fiducials.
+
+        Parameters
+        ----------
+        lpa_weight : float
+            Relative weight for LPA. The default value is 1.
+        nasion_weight : float
+            Relative weight for nasion. The default value is 10.
+        rpa_weight : float
+            Relative weight for RPA. The default value is 1.
+        %(verbose)s
```

Comment:
Methods that do not return anything should return self do we can chain operations. Don’t forget to update docstring when you update this.